src/Cake.Issues/IIssueComparer.cs

```diff
@@ -0,0 +1,135 @@
+        /// <remarks>
+        /// If <paramref name="compareOnlyPersistentProperties"/> is set to <c>true</c> the following
+        /// properties will be ignored while comparing the issue:
+        /// <list type="bullet">
+        /// <item>
+        /// <description><see cref="IIssue.ProjectFileRelativePath"/></description>
+        /// </item>
+        /// <item>
+        /// <description><see cref="IIssue.AffectedFileRelativePath"/></description>
+        /// </item>
+        /// <item>
+        /// <description><see cref="IIssue.Line"/></description>
+        /// </item>
+        /// </list>
+        /// </remarks>
+        public IIssueComparer(bool compareOnlyPersistentProperties)
+        {
+            this.compareOnlyPersistentProperties = compareOnlyPersistentProperties;
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(IIssue x, IIssue y)
+        {
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            if (x is null || y is null)
+            {
+                return false;
+            }
+
+            return
+                (this.compareOnlyPersistentProperties || x.ProjectFileRelativePath?.FullPath == y.ProjectFileRelativePath?.FullPath) &&
+                (x.ProjectName == y.ProjectName) &&
+                (this.compareOnlyPersistentProperties || x.AffectedFileRelativePath?.FullPath == y.AffectedFileRelativePath?.FullPath) &&
+                (this.compareOnlyPersistentProperties || x.Line == y.Line) &&
+                (x.MessageText == y.MessageText) &&
+                (x.MessageHtml == y.MessageHtml) &&
+                (x.MessageMarkdown == y.MessageMarkdown) &&
+                (x.Priority == y.Priority) &&
+                (x.PriorityName == y.PriorityName) &&
+                (x.Rule == y.Rule) &&
+                (x.RuleUrl?.ToString() == y.RuleUrl?.ToString()) &&
+                (x.ProviderType == y.ProviderType) &&
+                (x.ProviderName == y.ProviderName);
+        }
+
+        /// <inheritdoc/>
+        public int GetHashCode(IIssue obj)
+        {
+            if (obj is null)
+            {
+                return 0;
+            }
+
+            if (this.compareOnlyPersistentProperties)
+            {
+                return
+                    GetHashCode(
+                        obj.ProjectName,
+                        obj.MessageText,
+                        obj.MessageHtml,
+                        obj.MessageMarkdown,
+                        obj.Priority,
+                        obj.PriorityName,
+                        obj.Rule,
+                        obj.RuleUrl,
+                        obj.ProviderType,
+                        obj.ProviderName);
+            }
+            else
+            {
+                return
+                    GetHashCode(
+                        obj.ProjectFileRelativePath?.ToString(),
+                        obj.ProjectName,
```

Comment:
move this after `object.Line` to have the same order as in the equals method and the non persistent are grouped together

Reply:
Like it is now it is already in the same order as in the equals method

Reply:
@pascalberger yes was the same order, my mistake. But I would change the order in the `Equals` and `GetHashCode` so that the first 3 properties are the non persistent ones....